utils/core/src/serde/byte_writer.rs

```diff
@@ -141,9 +141,9 @@ impl ByteWriter for alloc::vec::Vec<u8> {
 // HELPER FUNCTIONS
 // ================================================================================================
 
-/// Returns the length of the value in vint64 encoding.
-fn encoded_len(value: u64) -> usize {
-    let zeros = value.leading_zeros() as usize;
+/// Returns the length of the usize value in vint64 encoding.
+pub(super) fn usize_encoded_len(value: usize) -> usize {
```

Comment:
Passing value as `usize` into here would cause issues on 32-bit systems (e.g., WASM). Let's revert this change.

Reply:
I see, good point!
Updated to cast the usize in the `get_size_hint` function to a u64 instead.